ma/maRefine.cc

```diff
@@ -345,14 +345,16 @@ void transferElements(Refine* r)
   Adapt* a = r->adapt;
   Mesh* m = a->mesh;
   SolutionTransfer* st = a->solutionTransfer;
-  int td = st->getTransferDimension();
-  for (int d = td; d <= m->getDimension(); ++d)
-    for (size_t i=0; i < r->toSplit[d].getSize(); ++i)
-      st->onRefine(r->toSplit[d][i],r->newEntities[d][i]);
-  td = a->shape->getTransferDimension();
+  int td = a->shape->getTransferDimension();
```

Comment:
@avinmoharana 

It seems that the order of the two loops here have been interchanged for the purpose of curved solutions transfer. Have you tested the effect of this on the rest of the code (i.e., linear mesh adapt solutions transfer)? We don't want to break other parts of the code!

Reply:
@mortezah 
Performed a linear adapt for the Laghos problem (mesh order = 1) and compared the coordinate field value at the mid edge location with a user-defined field initialized as the coordinate field. The values are identical.